public/components/query_compare/search_result/index.tsx

```
@@ -309,6 +309,57 @@ export const SearchResult = ({
     return { preTags: ['<em>'], postTags: ['</em>'] };
   };
 
+  const extractMemoryId = (queryString: string): string | null => {
+    try {
+      const query = JSON.parse(queryString);
+      return query?.query?.agentic?.memory_id || null;
+    } catch {
+      return null;
```

If the query string indeed has memory id, but the string is not a valid json string, should we throw the exception instead of assuming the id is null?